### PR TITLE
Improving TableConfig code

### PR DESCRIPTION
**Pull request by Tang8330**
## Changes

* Rewriting certain tests
* Made `GetTableConfig` into a receiver method

### CODE REVIEW

File: clients/shared/table_config.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ type GetTableCfgArgs struct {
 	DropDeletedColumns bool
 }
 
-func (g *GetTableCfgArgs) ShouldParseComment(comment string) bool {
```
Comment (Tang8330) [inline clients/shared/table_config.go:33]: ptr is unnecessary because the object isn't getting mutated.

File: lib/typing/snowflake.go [hunk 2]
```diff
@@ -6,11 +6,9 @@ import (
...
 )
 
-// https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.html
-
-func SnowflakeTypeToKind(snowflakeType string) KindDetails {
-	snowflakeType = strings.ToLower(snowflakeType)
-
+// snowflakeTypeToKind - Converts a Snowflake type to a KindDetails.
+// Following this spec: https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.html
+func snowflakeTypeToKind(snowflakeType string) KindDetails {
```
Comment (nathan-artie) [inline lib/typing/snowflake.go:11]: We can probably also get rid of the `strings.ToLower` here: https://github.com/artie-labs/transfer/pull/432/files#diff-e6f658cbda168f6e7a233f8bcf5807e9b37544c785787f3a1c4e3890c506b681R25

Comment (Tang8330) [reply lib/typing/snowflake.go:11]: Good catch!

File: lib/typing/bigquery.go [hunk 3]
```diff
@@ -9,9 +9,8 @@ import (
 
 const StreamingTimeFormat = "15:04:05"
 
-func BigQueryTypeToKind(rawBqType string) KindDetails {
+func bigQueryTypeToKind(rawBqType string) KindDetails {
 	bqType := rawBqType
-	bqType = strings.ToLower(bqType)
```
Comment (nathan-artie) [inline lib/typing/bigquery.go:14]: Can probably remove the `strings.ToLower` call on line 33.

**Review by nathan-artie — APPROVED** [review]

**Pull request merged**